sphinx/ext/collapse.py

```diff
@@ -0,0 +1,206 @@
+    ) -> list[nodes.Node]:
+        # Same as util.nested_parse_with_titles but try to handle nested
+        # sections which should be raised higher up the doctree.
+        memo = state.memo
+        surrounding_title_styles = memo.title_styles
+        surrounding_section_level = memo.section_level
+        memo.title_styles = []
+        memo.section_level = 0
+        try:
+            state.nested_parse(content, offset, node, match_titles=True)
+            title_styles = memo.title_styles
+            if (
+                not surrounding_title_styles
+                or not title_styles
+                or title_styles[0] not in surrounding_title_styles
+                or not state.parent
+            ):
+                # No nested sections so no special handling needed.
+                return [node]
+            # Calculate the depths of the current and nested sections.
+            current_depth = 0
+            parent = state.parent
+            while parent:
+                current_depth += 1
+                parent = parent.parent
+            current_depth -= 2
+            title_style = title_styles[0]
+            nested_depth = len(surrounding_title_styles)
+            if title_style in surrounding_title_styles:
+                nested_depth = surrounding_title_styles.index(title_style)
+            # Use these depths to determine where the nested sections should
+            # be placed in the doctree.
+            n_sects_to_raise = current_depth - nested_depth + 1
+            parent = state.parent
+            for _i in range(n_sects_to_raise):
+                if parent.parent:
+                    parent = parent.parent
+            parent.append(node)
+            return []
+        finally:
+            memo.title_styles = surrounding_title_styles
+            memo.section_level = surrounding_section_level
+
+
+# This constant can be modified by programmers that create their own
+# HTML builders outside the Sphinx core.
+HTML_5_BUILDERS = frozenset({'html', 'dirhtml'})
+
+
+class CollapsibleNodeTransform(SphinxPostTransform):
+    default_priority = 55
+
+    def run(self, **kwargs: Any) -> None:
+        self._process_collapsible_nodes(self.document, self.app.builder.name)
+
+    @staticmethod
+    def _process_collapsible_nodes(document: nodes.Node, builder_name: str) -> None:
+        """Filter collapsible and collapsible_summary nodes based on HTML 5 support."""
+        if builder_name in HTML_5_BUILDERS:
+            return
+
+        for summary_node in document.findall(summary):
+            summary_para = nodes.paragraph('', '', *summary_node)
+            summary_node.replace_self(summary_para)
+
+        for collapsible_node in document.findall(collapsible):
+            # A comment on the comment() nodes being inserted: replacing by [] would
+            # result in a "Losing ids" exception if there is a target node before
+            # the only node, so we make sure docutils can transfer the id to
```

Comment:
I guess this was copy-pasted from the `OnlyNodeTransform` 😉 
again I feel this is a horrible hack, that just serves to hide the actual issue rather than address it;
the id should not be lost without a warning, it should most probably be propagated to the next sibling

Reply:
> I guess this was copy-pasted from the `OnlyNodeTransform` 😉

Yes, I should update the text...

> again I feel this is a horrible hack, that just serves to hide the actual issue rather than address it;
> the id should not be lost without a warning, it should most probably be propagated to the next sibling

I think it is fine to loose the ID here. This function runs to remove the nodes from the doctree for unsupported builders. Does it make sense to keep an ID for a node that doesn't exist anymore?

As it happens, in every non-pathalogical case, children will exist, as the summary node is guaranteed by the ``.. collapse`` directive, and will always have children. We could therefore remove the ``comment()`` logic, I believe.

A